cmd/basecli/counter/counter.go

```diff
@@ -12,68 +12,72 @@ import (
 	btypes "github.com/tendermint/basecoin/types"
 )
 
+//CounterTxCmd is the CLI command to execute the counter
+//  through the appTx Command
 var CounterTxCmd = &cobra.Command{
 	Use:   "counter",
 	Short: "add a vote to the counter",
 	Long: `Add a vote to the counter.
 
 You must pass --valid for it to count and the countfee will be added to the counter.`,
-	RunE: doCounterTx,
+	RunE: counterTxCmd,
 }
 
 const (
-	CountFeeFlag = "countfee"
-	ValidFlag    = "valid"
+	flagCountFee = "countfee"
+	flagValid    = "valid"
 )
 
 func init() {
 	fs := CounterTxCmd.Flags()
 	bcmd.AddAppTxFlags(fs)
-	fs.String(CountFeeFlag, "", "Coins to send in the format <amt><coin>,<amt><coin>...")
-	fs.Bool(ValidFlag, false, "Is count valid?")
+	fs.String(flagCountFee, "", "Coins to send in the format <amt><coin>,<amt><coin>...")
+	fs.Bool(flagValid, false, "Is count valid?")
 }
 
-func doCounterTx(cmd *cobra.Command, args []string) error {
-	tx := new(btypes.AppTx)
+func counterTxCmd(cmd *cobra.Command, args []string) error {
 	// Note: we don't support loading apptx from json currently, so skip that
 
-	// read the standard flags
-	err := bcmd.ReadAppTxFlags(tx)
+	// Read the app-specific flags
+	name, data, err := getAppData()
 	if err != nil {
 		return err
 	}
 
-	// now read the app-specific flags
-	err = readCounterFlags(tx)
+	// Read the standard app-tx flags
```

Comment:
@rigelrozanski : possible improvement... 

If we want to get crazy with helper functions.  It seems that everything from here until BroadcastAppTx is the same on every app.

So in addition to BroadcastAppTx, we could have `PrepareAppTx(name, data) (*types.AppTx, error)` that creates the apptx.  Maybe even combine them into `PrepareAndBroadcastAppTx(...)`.... but then again, maybe that is going overboard and would obscure the funcionality